src/Airship/Internal/Helpers.hs

```diff
@@ -83,15 +83,16 @@ resourceToWai cfg routes resource404 =
   resourceToWaiT cfg id routes resource404
 
 -- | Given a 'RoutingSpec', a 404 resource, and a user state @s@, construct a WAI 'Application'.
-resourceToWaiT :: Monad m => AirshipConfig -> (forall a. m a -> IO a) -> RoutingSpec m () -> Resource m -> Wai.Application
+resourceToWaiT :: Monad m => AirshipConfig -> (m Wai.Response -> IO Wai.Response) -> RoutingSpec m () -> Resource m -> Wai.Application
```

Comment:
Unrelated, but I was/am tempted to add `Request` as an argument here. That would be useful if, for example, you wanted to add cross cutting concerns, like for example a security monad that needed the headers.